### PR TITLE
Update dev dependency: supertest

### PR DESCRIPTION
**Pull request by alxndrsn**
Resolves `npm` warning:

```
WARN deprecated superagent@3.8.3: Please upgrade to v7.0.2+ of superagent.  We have fixed numerous issues with streams, form-data, attach(), filesystem errors not bubbling up (ENOENT on attach()), and all tests are now passing.  See the releases tab for more information at <https://github.com/visionmedia/superagent/releases>.
```

### CODE REVIEW

File: test/integration/setup.js [hunk 1]
```diff
@@ -123,26 +123,43 @@ const authProxy = (token) => ({
     return (...args) => method.apply(target, args).set('Authorization', `Bearer ${token}`);
   }
 });
+// Add User-Agent header to all requests.  This was previously a default in
+// superagent, but was removed in 5.1.0.  We keep the value the same as before
+// we upgraded.
+// See: https://github.com/ladjs/superagent/pull/1495
+const uaProxy = {
```
[the remark this answers is not in the document]
Comment (matthew-white) [reply test/integration/setup.js]: Ah I see that tests still don't pass even now that #907 is merged. In test/assertions.js, how about we make the assertion on the user agent conditional? For example, in the `Submission` assertion:

```diff
-this.obj.userAgent.should.be.a.String();
+if (this.obj.userAgent != null) this.obj.userAgent.should.be.a.String();
```

We follow a similar pattern for other optional properties, for example, `updatedAt`.

I don't think we should always assert that the `userAgent` property exists. I think it'd be better for us to add specific tests about the user agent. I can file an issue about that if that'd be helpful. But I'm curious whether adding that null check to the assertion will help tests pass here. (The `Submission` assertion will need to change and also the `EntityDef` assertion.)

[the remark this answers is not in the document]
Comment (alxndrsn) [reply test/integration/setup.js]: Shame there's no `.or.is.Null()` - https://github.com/shouldjs/should.js/issues/183

**Comment by alxndrsn** [comment]
@matthew-white tests now passing :slightly_smiling_face:

File: test/integration/api/submissions.js [hunk 2]
```diff
@@ -4061,9 +4062,9 @@ one,h,/data/h,2000-01-01T00:06,2000-01-01T00:07,-5,-6,,ee,ff
               body[1].instanceName.should.equal('custom name');
 
               body[0].deviceId.should.equal('updateDevice');
-              body[0].userAgent.should.equal('central/tests');
               (body[1].deviceId == null).should.equal(true);
-              body[1].userAgent.should.equal('node-superagent/3.8.3');
+              body[0].userAgent.should.equal('central-tests/2');
+              body[1].userAgent.should.equal('central-tests/1');
```
Comment (alxndrsn) [inline test/integration/api/submissions.js:4067]: Does this order look correct?

Comment (matthew-white) [reply test/integration/api/submissions.js:4067]: I guess the more recent submission version is returned first? It seems consistent with how it was before.

**Review by matthew-white — APPROVED** [review]

**Pull request merged**